### DISCUSSION TITLE
Elasticsearch: Need possibility to return last document when using raw document query

### DISCUSSION THREAD

**Discussion by adeverteuil** [grafana/grafana#56253] [Observability]
**What would you like to be added**:

Group By clause for Raw Document queries. Only **First** and **Last** aggregation functions would make sense.

**Example use-case**:

We have an index that returns multiple documents in a Table panel displayed as Raw Documents.
Fields are `hostname`, `target`, `displaytext`, `loglevel`.
We want to display only the **last** message for a given `hostname` and `target`.

**What we tried**:

1. Set `Size: 1` in the Raw Document query. This doesn’t work because ES returns the **first** document by timestamp, and we want the last one.
2. Use the Date Histogram, group by `hostname` and `target`, aggregated by Last.
Aggregation functions don’t make sense for string data. This is why I opened this issue.

**Comment by torkelo** [comment]
raw document queries do not support aggregations, only query filters.

**Comment by torkelo** [comment]
Not sure how to support this. Maybe some new ES query filter feature could do it but not group by

**Comment by marefr** [comment]
I think what we need to allow is to sort by timestamp descending in raw document queries that in combination with selecting one document will solve this, right @adeverteuil ? I'm reopen this.

**Comment by adeverteuil** [comment]
Yes, aggregation doesn't make sense for raw documents, except selecting the first or last document for each Group By. Specifically, the customer wants to select the last document.

**Comment by dagwieers** [comment]
Exactly, we have the same need and now work around this by selection only a single data point (although we always get two). I think it makes more sense to have a "Current" or "Last" aggregation function that returns the most recent document.

This is not just useful for Group by, IMO. A "Current" aggregation function would work on string fields to return e.g. the latest "Health" string from e.g. an Elastic Heartbeat REST check. Now we work around this using Raw data, group by, 1 data point, but why query using Raw data if we only need a single field value of the most recent document?

**Comment by gabor** [comment]
i wonder if this is elasticsearch specific even. based on the initial comment, to me it seems we want to have something returned from elasticsearch, and then filter out some data from it in the browser.
isn't this job more suited for a Transform in the dashboards?

or, is there an elasticsearch query that can be built that returns the data in the form that is requested here?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
